ocs_ci/ocs/resources/cloud_manager.py

```diff
@@ -143,18 +158,19 @@ def internal_delete_uls(self, name):
            name (str): The Underlying Storage name to be deleted
 
         """
+        assert self.exec_uls_deletion(name)
         # Todo: rename client to resource (or find an alternative)
         sample = TimeoutSampler(
-            timeout=30, sleep=3, func=self.exec_uls_deletion,
-            name=name
+            timeout=60, sleep=6, func=self.verify_uls_exists,
+            uls_name=name
         )
-        if not sample.wait_for_func_status(result=True):
+        if not sample.wait_for_func_status(result=False):
```

Comment:
please make sure that this doesn't break existing usages

Reply:
We didn't change a function signature :)
We replaced `exec_uls_deletion` with `verify_uls_exists`. And instead of waiting for a `True` from the deletion function, we're now waiting for a `False` from the deletion verification function. Thus, no other existing usages.